m6-advanced-multithreading/t3-file-scanner-fjp/src/main/java/pl/mentoring/filescanner/DirStatisticsProvider.java

```diff
@@ -0,0 +1,72 @@
+package pl.mentoring.filescanner;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.RecursiveTask;
+
+public class DirStatisticsProvider implements Callable<DirStatistics> {
+
+    private final File dir;
+
+    public DirStatisticsProvider(File dir) {
+        if (!dir.isDirectory()) {
+            throw new IllegalArgumentException("Invalid directory path");
+        }
+        this.dir = dir;
+    }
+
+    @Override
+    public DirStatistics call() {
+        int processorsCount = Runtime.getRuntime().availableProcessors();
+        ForkJoinPool pool = new ForkJoinPool(processorsCount);
+
+        try {
+            return pool.invoke(new DirStatisticsWorker(dir));
+        } finally {
+            pool.shutdown();
+        }
+    }
+
+    private class DirStatisticsWorker extends RecursiveTask<DirStatistics> {
```

Comment:
Can you explain why it is not static class?

Reply:
There is no good reason for that. Changed to static